pkg/utils/pod/scheduling.go

```diff
@@ -53,10 +53,12 @@ func IsReschedulable(pod *v1.Pod) bool {
 // - Is an active pod (isn't terminal or actively terminating)
 // - Doesn't tolerate the "karpenter.sh/disruption=disrupting" taint
 // - Isn't a mirror pod (https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/)
+// - Doesn't have the "karpenter.sh/do-not-disrupt=true" annotation
 func IsEvictable(pod *v1.Pod) bool {
 	return IsActive(pod) &&
 		!ToleratesDisruptionNoScheduleTaint(pod) &&
-		!IsOwnedByNode(pod)
+		!IsOwnedByNode(pod) &&
+		IsDisruptable(pod)
```

Comment:
We're going to stop evicting `karpenter.sh/do-not-disrupt` pods in this change? That's a major breaking change and not something that I think we can do without the grace period mechanism that we've been talking about. We've told users that they can expect that there is a way to bail out of pods hanging nodes during drain by doing a `kubectl delete node`. I agree that we need to reverse this, but shouldn't do so without giving these same users a mechanism to gracefully delete their nodes with `kubectl delete node`

Reply:
This also strikes me as out of scope of this race condition fix -- something we want to do long-term, but probably not in the context of this bug fix

Reply:
Sounds good, I included it here as a mechanism to ensure that these pods aren't disrupted even if they slip through but I agree it can be decoupled.